tests/interpreter/cases/input/tests.yaml

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+cases:
+  - note: input-basic-1
+    data: {}
+    input: {
```

Comment:
Good idea.

Another thing, previously an `eval` call was enough to evaluate all modules, now it's replaced by:
```
interpreter.prepare_for_eval(Some(&schedule), &None)?;
let results = interpreter.eval_modules(&input, true)?;
```
maybe we should still have an `eval` function doing the two previous steps?

Also what about having a boolean tracking the call of `prepare_for_eval`, this way if an user call  `eval_modules` (or `eval_module` or `eval_rule_with_input` or `eval_rule` or `eval_query_snippet`) without  `prepare_for_eval` before then we warn him (or we do a `prepare_for_eval` transparently for him?)

Reply:
> You could tweak the yaml based test framework so that if input, and want_result are arrays

It would be good to have a `multiple_inputs` flag as well. Only if that flag is specified as true, then `input` and `want_result` will be interpreted as arrays.

> eval function doing the two previous steps

That sounds good.

> or we do a prepare_for_eval transparently for him

We could start off with this approach and revisit it later if needed.

Reply:
I was not able how to use a `multiple_inputs` flag to specify the type of `input` and `want_result` because a `Value` can be deserialized as an array, but maybe I am wrong and there is a way to do that?

Instead, I created an `Enum` and I modified each test case to specify what variant of the enum to expect.

Reply:
I finally feel that warning the user about the mandatory `prepare_for_eval` call is better than calling `prepare_for_eval `transparently for the user because `prepare_for_eval `expects the data as argument but not `eval_ ` functions and so it changes a bit the logic if `eval_` takes data also as an argument.